app/models/queue_item.rb

```diff
@@ -5,12 +5,27 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates_numericality_of :position, {only_integer: true}
+  
   def category_name
     category.name
   end
 
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
+
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else 
+      review = Review.new(creator: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
+
+  private
+  def review
+    @review ||= Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
You should use `find_by` instead of `where().first`.

HOMEWORK: Find out why `find_by` is better option and explain.